python/paddle/fluid/tests/unittests/test_index_select_op.py

```diff
@@ -131,5 +132,99 @@ def test_dygraph_api(self):
+
+    def test_dygraph(self):
+        paddle.disable_static()
+
+        inps_shape = [7, 6, 5, 4, 3]
+        array = np.arange(self.numel(inps_shape)).reshape(inps_shape)
+
+        index_shape = [7, 6, 5, 4, 3]
+        index = np.arange(self.numel(index_shape)).reshape(index_shape)
+        for i in range(len(inps_shape) - 1):
+
+            pt = paddle.to_tensor(array)
+            index_mod = index % (array.shape[0])
+            pindex = paddle.to_tensor(index_mod)
+            getitem_np = array[index_mod]
+            getitem_pp = pt[pindex]
+            self.assertTrue(np.array_equal(getitem_np, getitem_pp.numpy()))
+            array = array[0]
+            index = index[0]
+
+    def test_static_graph(self):
+        paddle.enable_static()
+        inps_shape = [1, 2, 3, 4, 5]
+        array = np.arange(
+            self.numel(inps_shape), dtype='float32').reshape(inps_shape)
+
+        index_shape = [7, 6, 5, 4, 3]
+        index = np.arange(self.numel(index_shape)).reshape(index_shape)
+
+        for i in range(len(inps_shape) - 1):
+            index_mod = index % (array.shape[0])
+            program = paddle.static.Program()
+
+            with paddle.static.program_guard(program):
+                x = paddle.static.data(
+                    name='x', shape=array.shape, dtype='float32')
+                if i % 2 == 0:
+                    index_dtype = 'int32'
+                    index_mod = index_mod.astype('int32')
+                else:
+                    index_dtype = 'int64'
+                    index_mod = index_mod.astype('int64')
+
+                index_p = paddle.static.data(
+                    name='index', shape=index.shape, dtype=index_dtype)
+
+                y = x[index_p]
+
+                place = paddle.fluid.CPUPlace(
+                ) if not paddle.fluid.core.is_compiled_with_cuda(
+                ) else paddle.fluid.CUDAPlace(0)
+
+                prog = paddle.static.default_main_program()
+                exe = paddle.static.Executor(place)
+
+                exe.run(paddle.static.default_startup_program())
+                fetch_list = [y.name]
+                getitem_pp = exe.run(
+                    prog,
+                    feed={x.name: array,
+                          index_p.name: index_mod},
+                    fetch_list=fetch_list)
+
+                getitem_np = array[index_mod]
+                self.assertTrue(np.array_equal(getitem_np, getitem_pp[0]))
+
+            array = array[0]
+            index = index[0]
+
+    def test_backward(self):
+        paddle.disable_static()
+        array = np.arange(4 * 3 * 2, dtype='float32').reshape([4, 3, 2])
+        index = [[1, 2], [0, 3]]
+
+        index_p = paddle.to_tensor(index)
+        pt = paddle.to_tensor(array, stop_gradient=False)
+
+        y = pt[index_p]
+        y = y * y
+        loss = y.sum()
+        loss.backward()
+        grad_torch = np.array([[[0., 2.], [4., 6.], [8., 10.]],
```

Comment:
rename this var

Reply:
revert the modification of index_select_op.